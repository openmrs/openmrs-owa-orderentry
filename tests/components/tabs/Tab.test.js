import React from 'react';
import Tab from '../../../app/js/components/tabs/Tab';

describe('`Tab` component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<Tab/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should trigger event when clicked', () => {
        const handleOnClick = jest.fn();
        const wrapper = shallow(<Tab onClick={handleOnClick}/>);
        const tab = wrapper.find('a');
        expect(handleOnClick.mock.calls.length).toBe(0);
        tab.simulate('click', {
            preventDefault: () => {},
          });
        expect(handleOnClick.mock.calls.length).toBe(1);
    });
});
