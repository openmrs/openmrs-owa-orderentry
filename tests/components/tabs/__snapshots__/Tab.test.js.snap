// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Tab\` component should render without crashing 1`] = `
<li
  className="ui-state-default ui-corner-top"
>
  <a
    className="ui-tabs-anchor"
    onClick={[Function]}
    role="presentation"
  />
</li>
`;
